tgui/packages/tgui/interfaces/Newscaster.js

```diff
@@ -0,0 +1,898 @@
+              content={photo ? ("Eject: " + photo.name) : "Insert Photo"}
+              tooltip={!photo && "Attach a photo to this wanted notice by holding the photograph in your hand."}
+              tooltipPosition="top"
+              onClick={() => act(photo ? 'eject_photo' : 'attach_photo')}
+            />
+            {!!photo && (
+              <PhotoThumbnail
+                name={"inserted_photo_" + photo.uid + ".png"}
+                float="right"
+              />
+            )}
+          </LabeledList.Item>
+          {isAdmin && (
+            <LabeledList.Item label="CentComm Lock" verticalAlign="top">
+              <Button
+                selected={adminLocked}
+                icon={adminLocked ? "lock" : "lock-open"}
+                content={adminLocked ? "On" : "Off"}
+                tooltip="Locking this wanted notice will make it editable by nobody but CentComm officers."
+                tooltipPosition="top"
+                onClick={() => setAdminLocked(!adminLocked)}
+              />
+            </LabeledList.Item>
+          )}
+        </LabeledList>
+      </Box>
+      <Button.Confirm
+        disabled={!wanted}
+        icon="eraser"
+        color="danger"
+        content="Clear"
+        position="absolute"
+        right="7.25rem"
+        bottom="-0.75rem"
+        onClick={() => {
+          act("clear_wanted_notice");
+          modalClose(context);
+          // Clean up
+          deleteLocalState(context, "author", "name", "description", "admin_locked");
+        }}
+      />
+      <Button.Confirm
+        disabled={author.trim().length === 0 || name.trim().length === 0 || description.trim().length === 0}
+        icon="check"
+        color="good"
+        content="Submit"
+        position="absolute"
+        right="1rem"
+        bottom="-0.75rem"
+        onClick={() => {
+          modalAnswer(context, modal.id, "", {
+            author: author,
+            name: name.substr(0, 127),
+            description: description.substr(0, 511),
+            admin_locked: adminLocked ? 1 : 0,
+          });
+          // Clean up
+          deleteLocalState(context, "author", "name", "description", "admin_locked");
+        }}
+      />
+    </Section>
+  );
+};
+
+// TODO: make this common
+const timeAgo = (ref_time, now_time) => {
+  if (ref_time > now_time) {
+    return "in the future";
+  }
+
+  // deciseconds -> seconds
+  ref_time = ref_time / 10;
+  now_time = now_time / 10;
+
+  const diff = now_time - ref_time;
+  if (diff > 3600) {
+    const hours = Math.round(diff / 3600);
+    return hours + " hour" + (hours === 1 ? "" : "s") + " ago";
+  } else if (diff > 60) {
+    const mins = Math.round(diff / 60);
+    return mins + " minute" + (mins === 1 ? "" : "s") + " ago";
+  } else {
+    const secs = Math.round(diff);
+    return secs + " second" + (secs === 1 ? "" : "s") + " ago";
+  }
+
+  return "just now";
+};
+
+export const Newscaster = (properties, context) => {
```

Comment:
Any reason this is all the way down instead of at the top?

Reply:
I typically put constants first but I'll definitely move the modal body overrides at the end of the file.